src/components/draggable/draggable-container.component.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState, useRef, useMemo } from "react";
+import { DndProvider, useDrop } from "react-dnd";
+import { HTML5Backend } from "react-dnd-html5-backend";
+
+import invariant from "invariant";
+import { filterStyledSystemMarginProps } from "../../style/utils";
+import DraggableItem from "./draggable-item.component";
+import {
+  StyledDraggableContainer,
+  StyledDraggableContainerProps,
+} from "./draggable-item.style";
+
+export interface DraggableContainerProps extends StyledDraggableContainerProps {
+  /** Callback fired when order is changed */
+  getOrder?: (
+    draggableItemIds?: (string | number | undefined)[],
+    movedItemId?: string | number | undefined
+  ) => void;
+  /**
+   * The content of the component
+   *
+   * `<DraggableItem />` is required to make `Draggable` works
+   */
+  children?: React.ReactNode;
+}
+
+interface DropItem {
+  id: number | string;
+  originalIndex: number;
+}
+
+const DropTarget = ({
+  children,
+  getOrder,
+  ...rest
+}: { getOrder: (movedItemId?: string | number | undefined) => void } & Omit<
+  DraggableContainerProps,
+  "getOrder"
+>) => {
+  const [, drop] = useDrop({
+    accept: "draggableItem",
+    drop(item: DropItem) {
+      // istanbul ignore else
+      if (getOrder) {
+        getOrder(item?.id);
+      }
+    },
+  });
+
+  return (
+    <StyledDraggableContainer ref={drop} {...rest}>
+      {children}
+    </StyledDraggableContainer>
+  );
+};
+
+const DraggableContainer = ({
+  children,
+  getOrder,
+  ...rest
+}: DraggableContainerProps): JSX.Element => {
+  const [draggableItems, setDraggableItems] = useState(
+    React.Children.toArray(children)
+  );
+
+  const hasProperChildren = useMemo(() => {
+    let hasDraggableItemChildren = true;
+
+    React.Children.toArray(children).forEach((child) => {
+      if (
+        typeof child === "string" ||
+        (React.isValidElement(child) &&
+          (child.type as React.FunctionComponent).displayName !==
+            DraggableItem.displayName)
+      ) {
+        hasDraggableItemChildren = false;
+      }
+    });
+
+    return hasDraggableItemChildren;
+  }, [children]);
+
+  // `<DraggableItem />` is required to make `Draggable` work
+  invariant(
+    hasProperChildren,
+    `\`${DraggableContainer.displayName}\` only accepts children of type \`${DraggableItem.displayName}\`.`
+  );
+
+  const isFirstRender = useRef(true);
+
+  useEffect(() => {
+    if (!isFirstRender.current) {
+      setDraggableItems(React.Children.toArray(children));
+    } else {
+      isFirstRender.current = false;
+    }
+  }, [children]);
+
+  const findItem = (id: string | number) => {
+    const draggableItem = draggableItems.filter((item: any) => {
```

Comment:
We need to type this better than `any`, I'm surprised the linter hasn't complained here

Reply:
This should be fine to use unless I've missed something

```return React.isValidElement(item) && `${item.props.id}` === id;```